src/app/feed/feed-post/feed-post.component.ts

```diff
@@ -392,8 +394,20 @@ export class FeedPostComponent implements OnInit {
       });
   }
 
-  getEmbedVideoURLForPostContent(): string {
-    return VideoUrlParserService.getEmbedVideoURL(this.postContent.PostExtraData["EmbedVideoURL"]);
+  getEmbedVideoURLForPostContent(): any {
+    return this.constructedEmbedVideoURL;
+  }
+
+  async setEmbedVideoURLForPostContent(): Promise<void> {
+    this.constructedEmbedVideoURL = await VideoUrlParserService.getEmbedVideoURL(
```

Comment:
Let's not use the async / await pattern here. Return a promise for `VideoUrlParserService.getEmbedVideoURL` and then call `VideoUrlParserService.getEmbedVideoURL().subscribe(res => this.constructEmbedVideoUrl = res`